### ISSUE TITLE
Buildomat: Target for Oxide VMs

### ISSUE THREAD

**Issue by smklein** [oxidecomputer/buildomat#13]
We currently have a "factory" for AWS, and one for lab infrastructure within https://github.com/oxidecomputer/buildomat/tree/main/factory

We should add a third factory for "oxide", which appears similar to the AWS interface, but which creates VMs on an Oxide rack instead.

**Comment by davepacheco** [comment]
I assume that this would use the Rust SDK that's part of the [CLI](https://github.com/oxidecomputer/cli).  Is that right?  If so somebody (maybe @karencfv?) may want to make sure we've implemented whatever pieces of the SDK would be needed here.

**Comment by davepacheco** [comment]
Err, I think I erroneously concluded the above based on discussions long ago.  I'm not sure if there is a Rust SDK.  Maybe we just want to use a progenitor client.  I'll ping folks offline to discuss.

**Comment by smklein** [comment]
Yeah, we're currently making use of a progenitor-based one here: https://github.com/oxidecomputer/omicron/blob/main/oxide-client/src/lib.rs

(that's used inside nexus for end-to-end tests)
